jina/hubble/hubio.py

```diff
@@ -0,0 +1,97 @@
+"""Module for wrapping Jina Hub API calls."""
+
+import os
+import argparse
+import json
+from pathlib import Path
+
+from ..helper import (
+    colored,
+    get_full_version,
+)
+from ..logging.logger import JinaLogger
+from ..logging.profile import TimeContext
+from .helper import archive_package
+
+
+HUBBLE_REGISTRY = os.environ.get('HUBBLE_REGISTRY', 'https://hubble.jina.ai')
+
+
+class HubIO:
+    """:class:`HubIO` provides the way to interact with Jina Hub registry.
+    You can use it with CLI to package a directory into a Jina Hub and publish it to the world.
+    Examples:
+        - :command:`jina hub push my_executor/` to push the executor package to Jina Hub
+        - :command:`jina hub pull UUID8` to download the executor identified by UUID8
+    """
+
+    def __init__(self, args: 'argparse.Namespace'):
+        """Create a new HubIO.
+        :param args: arguments
+        """
+        self.logger = JinaLogger(self.__class__.__name__, **vars(args))
+        self.args = args
+
+    def push(self) -> None:
+        """Push the executor pacakge to Jina Hub."""
+
+        import requests
+        import hashlib
+
+        is_public = True
+        if self.args.private:
+            is_public = False
+            self.args.public = False
+
+        pkg_path = Path(self.args.path)
+        if not pkg_path.exists():
+            self.logger.critical(
+                f'The folder "{self.args.path}" does not exist, can not push'
+            )
+            raise FileNotFoundError(
+                f'The folder "{self.args.path}" does not exist, can not push'
+            )
+
+        try:
+            # archive the executor package
+            with TimeContext(f'archiving executor at {self.args.path}', self.logger):
+                md5_hash = hashlib.md5()
+                bytesio = archive_package(pkg_path)
+                content = bytesio.getvalue()
+                md5_hash.update(content)
+
+                md5_digest = md5_hash.hexdigest()
+
+            # upload the archived package
+            meta, env = get_full_version()
+            form_data = {
+                'meta': json.dumps(meta),
+                'env': json.dumps(env),
+                'is_public': is_public,
+                'md5sum': md5_digest,
+                'force': self.args.force,
+                'secret': self.args.secret,
+            }
+
+            # upload the archived executor to Jina Hub
+            upload_url = HUBBLE_REGISTRY + '/v1/executors/push'
```

Comment:
put it in the header after `HUBBLE_REGISTRY` for better visibility and maintainbility